compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/Transforms.h

```diff
@@ -39,4 +40,28 @@ collapseOpIterationDims(AttentionOp op,
                         ArrayRef<ReassociationIndices> foldedIterationDims,
                         RewriterBase &rewriter);
 
+/// Rewrite an input rfft op (from an upstream dialect) into linalg_ext.fft.
+/// Only applies to power of 2 FFT sizes. Return real and imaginary tensor
+/// values or failure.
+///
+/// op: merely used for getting the Location and for reporting match failures.
+/// operand: input operand to FFT, must be of type RankedTensorType and with
+///   static shapes for the pattern to apply.
+/// fftLength: size of input dimension
+///   along which FFT is computed, must be power of 2 for the pattern to apply
+///
+/// Example usage from Torch to LinalgExt:
+///
+/// Value builtinCast =
+/// rewriter.create<torch::TorchConversion::ToBuiltinTensorOp>(
+///            loc,
+///            cast<torch::Torch::ValueTensorType>(self.getType())
+///                .toBuiltinTensor(),
+///            self);
+/// int64_t fftLength = inputShape[dim]; // where dim is FFT dimension
+/// auto res = rewriteFft(op, builtinCast, fftLength, rewriter);
```

Comment:
I was thinking more about providing an IR example and explaining what `op`, `operand`, and `fftLength` correspond to in the IR. Something like:
```
/// As an example, `%arg0` is the |operand|, the `torch.aten.fft_rfft` is passed as |op|, and `%int2` is passed as the static |fftLength| for the below snippet.
///
/// %int2 = torch.constant.int 2 : !torch.int
/// %out = torch.aten.fft_rfft %arg0, %none, %int2, %none : !torch.vtensor<[3,8,16],f32>, !torch.none, !torch.int, !torch.none -> !torch.vtensor<[3,8,9],complex<f32>>
```